nautobot/extras/webhooks.py

```diff
@@ -2,12 +2,46 @@
 from django.utils import timezone
 
 from nautobot.utilities.api import get_serializer_for_model
-from nautobot.extras.models import Webhook
+from nautobot.extras.models import Webhook, ObjectChange
 from nautobot.extras.registry import registry
 from nautobot.extras.tasks import process_webhook
+from nautobot.utilities.utils import shallow_compare_dict
 from .choices import ObjectChangeActionChoices
 
 
+def get_snapshots(instance, action):
+    prechange = None
+
+    if action != ObjectChangeActionChoices.ACTION_CREATE:
+        changed_object_type = ContentType.objects.get_for_model(instance)
+        most_recent_changes = ObjectChange.objects.filter(
+            changed_object_type=changed_object_type, changed_object_id=instance.id
+        )[:2]
+        prechange = most_recent_changes[1].object_data_v2 if most_recent_changes.count() > 1 else None
+
+    serializer_class = get_serializer_for_model(instance.__class__)
+
+    postchange = (
+        serializer_class(instance, context={"request": None}).data
+        if action != ObjectChangeActionChoices.ACTION_DELETE
+        else None
+    )
```

Comment:
Maybe I'm missing something here, but couldn't we use `most_recent_changes[0].object_data_v2` as the postchange data, instead of regenerating it?

If for some reason that doesn't work, as a fallback we could at least pass `serializer` as a parameter from `enqueue_webhooks` to this function rather than needing to recreate it here